lib/ansible/plugins/filter/core.py

```diff
@@ -283,6 +283,28 @@ def to_uuid(string, namespace=UUID_NAMESPACE_ANSIBLE):
     return to_text(uuid.uuid5(uuid_namespace, to_native(string, errors='surrogate_or_strict')))
 
 
+def is_uuid(a, version=None):
+    '''check if input is uuid'''
```

Comment:
Instead of using a regex, it would be much more simple to:

```python
import uuid

try:
    UUID(to_text(a, errors='surrogate_or_strict'))
except ValueError:
    return False
else:
    return True
```

Reply:
Yes, I had originally use a try statement, but changed it to a regex due to some possible edge cases, where the some values are not flagged as invalid, but is converted to a valid UUID.

Reference:
https://stackoverflow.com/questions/11384589/what-is-the-correct-regex-for-matching-values-generated-by-uuid-uuid4-hex

I didn't confirm this, but decided to avoid try for that reason.